ads/adfox.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, validateData} from '../3p/3p';
+import {yandex} from './yandex';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function adfox(global, data) {
+
+  validateData(data, [
+    'adfoxParams',
+    'ownerId',
+  ]);
+
+  loadAdFox(global, () => initAdFox(global, data));
+}
+
+/**
+ * @param {!Window} global
+ * @param {!Function} cb
+ */
+function loadAdFox(global, cb) {
+  loadScript(global, 'https://yastatic.net/pcode/adfox/loader.js', cb);
+}
+
+/**
+ * @param {!Window} global
+ * @param {string} name
+ */
+function createContainer(global, name) {
+  const d = global.document.createElement('div');
```

Comment:
jfyi
When I tried to use the method `createElementWithAttributes` from `src/dom`, travis did not pass the test with error `Found forbidden: "win.open(" in dist.3p/current/integration.js:9897:11`.
It seems that this is an incorrect behavior.